File: win-prog/ProgWinVB/String Theory/StringBuilderAppend/StringBuilderAppend.vb
'----------------------------------------------------
' StringBuilderAppend.vb (c) 2002 by Charles Petzold
'----------------------------------------------------
Imports System
Imports System.Text

Module StringBuilderAppend
    Const iIterations As Integer = 10000

    Sub Main()
        Dim dt As DateTime = DateTime.Now
        Dim sb As New StringBuilder()
        Dim i As Integer

        For i = 1 To iIterations
            sb.Append("abcdefghijklmnopqrstuvwxyz" & vbCrLf)
        Next i
        Dim str As String = sb.ToString()

        Console.WriteLine(DateTime.op_Subtraction(DateTime.Now, dt))
    End Sub
End Module

File: win-prog/ProgWinCS/String Theory/StringBuilderAppend/StringBuilderAppend.cs
//--------------------------------------------------
// StringBuilderAppend.cs © 2001 by Charles Petzold
//--------------------------------------------------
using System;
using System.Text;

class StringBuilderAppend
{
     const int iIterations = 10000;

     public static void Main()
     {
          DateTime      dt = DateTime.Now;
          StringBuilder sb = new StringBuilder();

          for (int i = 0; i < iIterations; i++)
               sb.Append("abcdefghijklmnopqrstuvwxyz\r\n");

          string str = sb.ToString();

          Console.WriteLine(DateTime.Now - dt);
     }
}